README.md

```diff
@@ -101,3 +101,100 @@ The command line interface also gives a quick overview of the different options
 
 3. By default HTTP requests to the hub timeout after 60 seconds. This can be
    changed by setting the `JUPYTERHUB_REQUEST_TIMEOUT` environment variable.
+
+## How it works
+
+jupyterhub-idle-culler culls user servers using JupyterHub's REST API
+([/users/{name}/server](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#operation--users--name--server-delete)
+or
+[/users/{name}/servers/{server_name}](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#operation--users--name--servers--server_name--delete)),
+and makes the culling decisions based on its configuration and what JupyterHub
+reports about the user servers via its REST API
+[(/users)](https://jupyterhub.readthedocs.io/en/stable/_static/rest-api/index.html#path--users)
+where user servers' `last_activity` is reported back.
+
+The `last_activity` that JupyterHub reports is the most recent summary of
+information updated at a regular interval via the [`update_last_activity`
+function](https://github.com/jupyterhub/jupyterhub/blob/1.4.2/jupyterhub/app.py#L2646)
+that combines two sources of information.
+
+1. **The proxy's routes data**
+
+   The `update_last_activity` function will [ask the
+   proxy](https://jupyterhub.readthedocs.io/en/stable/reference/proxy.html#retrieving-routes)
+   for the active routes and collects associated `last_activity` data if it is
+   available.
+
+   `last_activity` data for routes will be available when using
+   [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy#readme)
+   as JupyterHub does by default, but if for example
+   [traefik-proxy](https://github.com/jupyterhub/traefik-proxy#readme) is used,
+   no such data will be available.
+
+2. **The user server's activity reports**
+
+   The `update_last_activity` function also reads JupyterHub's database that
+   keeps state about servers `last_activity`. These database records are updated
+   whenever a server notifies JupyterHub about activity, as they are
+   responsibility to do.
+
+   As the servers need to be aware of JupyterHub to notify it about activity,
+   they are supposed to be started via the
```

Comment:
This is the default, right? JupyterHub doesn't work without `jupyterhub-singleuser` unless explicitly configured to otherwise with tricks (like we do for binder). There's also possible complication with `jupyter-labhub` and I don't even know what should be used when.

Reply:
Do you mean is it the default for JupyterHub to start servers with jupyterhub-singleuser?

I've spoken with Min about the binder exception, and he wanted me to avoid mentioning that btw as Min considers it a discouraged practice that we shouldn't end up help users do.

Reply:
yeah I agree. It's default for JupyterHub to start servers with `jupyterhub-singleuser`. The text here makes it sound like it is one of the supported options among many, and you need to use this if you want idle culling. But you need to use it regardless, so some tweaking of the text is probably all that's needed here.